src/Simplify.elm

```diff
@@ -5127,6 +5137,204 @@ subAndCmdBatchChecks moduleName { lookupTable, parentRange, fnRange, firstArg }
 
 
 
+-- HTML.ATTRIBUTES
+
+
+htmlAttributesClassListFalseElementError : { message : String, details : List String }
+htmlAttributesClassListFalseElementError =
+    { message = "In a Html.Attributes.classList, a tuple paired with False can be removed"
+    , details = [ "You can remove the tuple list element where the second part is False." ]
+    }
+
+
+htmlAttributesClassListChecks : CheckInfo -> List (Error {})
+htmlAttributesClassListChecks checkInfo =
+    let
+        listArg : Node Expression
+        listArg =
+            checkInfo.firstArg
+
+        getTupleWithSecond : Bool -> Node Expression -> Maybe { first : Node Expression, range : Range }
+        getTupleWithSecond bool expression =
+            case Node.value expression of
+                Expression.TupledExpression (first :: second :: []) ->
+                    if getBool bool checkInfo.lookupTable second then
+                        Just { range = Node.range expression, first = first }
+
+                    else
+                        Nothing
+
+                _ ->
+                    Nothing
+
+        singleElementListChecks : { a | element : Node Expression } -> List (Error {})
+        singleElementListChecks single =
+            case getTupleWithSecond True single.element of
+                Just tuple ->
+                    singleTrueChecks tuple
+
+                Nothing ->
+                    case getTupleWithSecond False single.element of
+                        Just _ ->
+                            [ Rule.errorWithFix htmlAttributesClassListFalseElementError
+                                checkInfo.fnRange
+                                [ Fix.replaceRangeBy (Node.range listArg) listCollection.emptyAsString ]
+                            ]
+
+                        Nothing ->
+                            []
+
+        singleTrueChecks : { a | first : Node Expression } -> List (Error {})
+        singleTrueChecks { first } =
+            [ Rule.errorWithFix
+                { message = "Html.Attributes.classList with a single tuple paired with True can be replaced with Html.Attributes.class"
+                , details = [ "You can replace this call by Html.Attributes.class with the String from the single tuple list element." ]
+                }
+                checkInfo.fnRange
+                (keepOnlyFix { parentRange = Node.range listArg, keep = Node.range first }
+                    ++ [ Fix.replaceRangeBy checkInfo.fnRange "Html.Attributes.class"
+                       ]
+                )
+            ]
+    in
+    case getListLiteral listArg of
+        Just (single :: []) ->
+            singleElementListChecks { element = single }
+
+        Just nonSingletonList ->
+            case findMapSurrounding (getTupleWithSecond False) nonSingletonList of
```

Comment:
The `tryMap` argument to `findMapSurrounding` is always `getTupleWithSecond`. You can probably hardcode that function in the implementation? Unless you think you'll use this for other simplifications?

Also, I'm trying to find a better name for this function but haven't figured one out yet (and depends on whether you agree on hardcoding `getTupleWithSecond` in the function)

Reply:
> you think you'll use this for other simplifications?

Think for example removing `Nothing`s in `filterMap`

Reply:
Alright, yeah that makes sense as a use-case 👍